### PR TITLE
Library update

### PR DESCRIPTION
**Pull request by eliaslecomte**
* Fix iOS open wifi path
* Cleanup the android code
* Update the readme
* Update javadoc
* Fix bugs on Android
* Make sure you can connect to a wifi network without scanning for it (which requires location permission starting Android 6).
* Move to AndroidX
* Update gradle.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ android {
     buildToolsVersion '29.0.2'
 
     defaultConfig {
-        minSdkVersion 19
+        minSdkVersion 21
```
Comment (JuanSeBestia) [inline android/build.gradle]: [Curiosity] Do we have a reason to upgrade `minSdkVersion`?

File: README.md [hunk 2]
```diff
@@ -66,4 +91,53 @@ WifiManager.getCurrentWifiSSID()
 	console.log('Cannot get current SSID!')
 })
 ```
-  
\ No newline at end of file
+
+## API
+
+_The api documentation is in progress._
+
```
Comment (JuanSeBestia) [inline README.md]: We should add for which platform they are supported, in IOS `getCurrentWifiSSID`,` connectToSSID`, `connectToProtectedSSID` and` disconnectFromSSID` are only implemented at the moment

File: README.md [hunk 3]
```diff
@@ -66,4 +91,53 @@ WifiManager.getCurrentWifiSSID()
...
+Type: `string`
+The SSID of the wifi network to connect with.
+
+#### password
+Type: `string`
+The password of the wifi network to connect with.
+
+#### isWeb
+Type: `boolean`
+Used on iOS. FIXME: why?
```
Comment (JuanSeBestia) [inline README.md]: isWep*
Look in doc not has interface with passphrase and without isWep
https://developer.apple.com/documentation/networkextension/nehotspotconfiguration

**Comment by JuanSeBestia** [comment]
We have conflicts :(, please resolve them

**Comment by JuanSeBestia** [comment]
I'm very happy with your contribution

**Comment by Rapsssito** [comment]
We need this ASAP!

File: android/build.gradle (hunk 1 above)
Comment (eliaslecomte) [reply android/build.gradle]: Android 21 (lollipop) was released in 2014. Hardly anyone is still using this. Targeting a lower android version had other warnings in the code that had to be fixed.

File: README.md (hunk 2 above)
Comment (eliaslecomte) [reply README.md]: I agree, I would add something like [android, ios] behind every method to state which platforms support what.

File: README.md (hunk 3 above)
Comment (eliaslecomte) [reply README.md]: Tbh I would not support connect to Wep networks at all :-D. They can be hacked in seconds.

**Comment by eliaslecomte** [comment]
I've fixed the merge conflicts (I think).

File: android/build.gradle (hunk 1 above)
Comment (JuanSeBestia) [reply android/build.gradle]: Thanks

**Comment by JuanSeBestia** [comment]
Doing manual testing...

**Comment by bitcrumb** [comment]
Thanks @eliaslecomte for your effort!

**Comment by thibmaek** [comment]
🐹 @eliaslecomte 🌶

**Pull request merged**